core/example/tutorial/14_unmanaged_view/parallel_for_example.cpp

```diff
@@ -0,0 +1,137 @@
+/****************************************************************************
+ * Copyright (c) 2018-2019 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Cabana_Core.hpp>
+#include <Cabana_Sort.hpp>
+
+#include <cassert>
+
+//---------------------------------------------------------------------------//
+// parallel for using an unmanaged view.
+//---------------------------------------------------------------------------//
+void parallelForExample()
+{
+    /* Declare general run parameters */
+    using MemorySpace = Kokkos::HostSpace;
+    using ExecutionSpace = Kokkos::DefaultHostExecutionSpace;
+    using DeviceType = Kokkos::Device<ExecutionSpace,MemorySpace>;
+    const int VectorLength = 8;
+
+    /*
+     * Declare Underlying data type
+    */
+
+    // Regular class form
+    class Data {
+        public:
+            double a[VectorLength];
+            double b[VectorLength];
+    };
+
+    // Cabana MemberTypes form:
+    using DataTypes = Cabana::MemberTypes<double,double>;
+
+
+    /*
+     * Create the AoSoA.
+    */
+    const int num_tuple = 128;
+    const int num_soa = 128/8; // Be careful with the int division..
+
+    Data* local_data = new Data[num_tuple];
```

Comment:
Do you mean to allocate on the free store?  Since the size is known at compile time, why not having `local_data` on the stack?  This way, we don't encourage our users to do `new` and `delete` :)

Reply:
In general a user that wants to wrap an unmanaged view around their data likely does not know the sizes at compile time. Therefore `new` or `malloc` is the correct choice for this example in terms of memory allocation.

Reply:
Agreed. I can (optionally) pull it from `argv` if we really don't want it to be known at compile time, but I think dynamic is the more useful example